test/integration/scheduler/framework_test.go

```diff
@@ -1326,39 +1329,47 @@ func TestCoSchedulingWithPermitPlugin(t *testing.T) {
 		permitPlugin.waitAndAllowPermit = test.waitAllow
 
 		// Create two pods.
-		waitingPod, err := createPausePod(testCtx.clientSet,
-			initPausePod(testCtx.clientSet, &pausePodConfig{Name: "waiting-pod", Namespace: testCtx.ns.Name}))
+		podA, err := createPausePod(testCtx.clientSet,
+			initPausePod(testCtx.clientSet, &pausePodConfig{Name: "pod-a", Namespace: testCtx.ns.Name}))
 		if err != nil {
-			t.Errorf("Error while creating the waiting pod: %v", err)
+			t.Errorf("Error while creating the first pod: %v", err)
 		}
-		signallingPod, err := createPausePod(testCtx.clientSet,
-			initPausePod(testCtx.clientSet, &pausePodConfig{Name: "signalling-pod", Namespace: testCtx.ns.Name}))
+		podB, err := createPausePod(testCtx.clientSet,
+			initPausePod(testCtx.clientSet, &pausePodConfig{Name: "pod-b", Namespace: testCtx.ns.Name}))
 		if err != nil {
-			t.Errorf("Error while creating the signalling pod: %v", err)
+			t.Errorf("Error while creating the second pod: %v", err)
 		}
 
 		if test.waitReject {
-			if err = waitForPodUnschedulable(testCtx.clientSet, waitingPod); err != nil {
+			if err = waitForPodUnschedulable(testCtx.clientSet, podA); err != nil {
 				t.Errorf("test #%v: Didn't expect the waiting pod to be scheduled. error: %v", i, err)
 			}
-			if err = waitForPodUnschedulable(testCtx.clientSet, signallingPod); err != nil {
+			if err = waitForPodUnschedulable(testCtx.clientSet, podA); err != nil {
 				t.Errorf("test #%v: Didn't expect the signalling pod to be scheduled. error: %v", i, err)
 			}
+			if (permitPlugin.waitingPod == podA.Name && permitPlugin.rejectingPod == podB.Name) ||
```

Comment:
I think we don't need to check the internals

Reply:
I think it is useful to know if test passed because anticipated events happened. I can see why both pods could be scheduled or unscheduled for other reasons than one waiting and other allowing or rejecting resulting in false positives.

Reply:
Ideally, we should be checking user-visible effects. For example, if a pod fails to schedule due to being rejected while waiting, then that message will be part of the scheduling failure event.

Capturing that is possible (example: https://github.com/kubernetes/kubernetes/pull/88526/files#diff-cdc3caeca8898a0c12df164e679baef3R598). But this can be done as a follow up if you wish to work on it. This is good for now, provided that we have experimental proof that it's working.